powersimdata/data_access/execute_list.py

```diff
@@ -107,39 +105,36 @@ def get_status(self, scenario_id):
             raise Exception(f"Scenario not found in execute list, id = {scenario_id}")
 
     def add_entry(self, scenario_info):
-        """Adds scenario to the execute list file on server.
+        """Add entry to execute list
 
         :param collections.OrderedDict scenario_info: entry to add
         """
-        print("--> Adding entry in execute table on server")
-        entry = "%s,created" % scenario_info["id"]
-        command = "echo %s >> %s" % (entry, self._server_path)
-        err_message = "Failed to update %s on server" % self._EXECUTE_LIST
-        _ = self._execute_and_check_err(command, err_message)
+        scenario_id = int(scenario_info["id"])
+        return self.set_status(scenario_id, "created")
 
-    def update_execute_list(self, status, scenario_info):
-        """Updates status in execute list file on server.
+    @verify_hash
+    def set_status(self, scenario_id, status):
+        """Set the scenario status
 
-        :param str status: execution status.
-        :param collections.OrderedDict scenario_info: entry to update
+        :param int/str scenario_id: the scenario id
+        :param str status: the new status
+        :return: (*pandas.DataFrame*) -- the updated data frame
         """
-        print("--> Updating status in execute table on server")
-        options = "-F, -v OFS=',' -v INPLACE_SUFFIX=.bak -i inplace"
-        # AWK parses the file line-by-line. When the entry of the first column is equal
-        # to the scenario identification number, the second column is replaced by the
-        # status parameter.
-        program = "'{if($1==%s) $2=\"%s\"};1'" % (scenario_info["id"], status)
-        command = "awk %s %s %s" % (options, program, self._server_path)
-        err_message = "Failed to update %s on server" % self._EXECUTE_LIST
-        _ = self._execute_and_check_err(command, err_message)
-
-    def delete_entry(self, scenario_info):
+        table = self.get_execute_table()
+        table.loc[int(scenario_id), "status"] = status
+
+        print(f"-->  Setting status={status} in execute table on server")
+        return table
+
+    @verify_hash
+    def delete_entry(self, scenario_id):
```

Comment:
I noticed we don't actually have code calling the delete method (ditto for the scenario list), but figured it might be useful to keep?

Reply:
It should be called in the `powersimdata.scenario.delete` module

Reply:
Well that makes sense. Not sure how I missed it

Reply:
`grep -R "delete_entry" powersimdata`